eslint/babel-eslint-tests/test/helpers/verifyAndAssertMessages.js

```diff
@@ -0,0 +1,70 @@
+import path from "path";
+import eslint from "eslint";
+import unpad from "dedent";
+import * as parser from "@babel/eslint-parser";
+
+export default function verifyAndAssertMessages(
+  code,
+  rules = {},
+  expectedMessages = [],
+  sourceType,
+  overrideConfig,
+) {
+  const linter = new eslint.Linter();
+  linter.defineParser("@babel/eslint-parser", parser);
+
+  const messages = linter.verify(unpad(`${code}`), {
+    parser: "@babel/eslint-parser",
+    rules,
+    env: {
+      node: true,
+      es6: true,
+    },
+    ...overrideConfig,
+    parserOptions: {
+      sourceType,
+      requireConfigFile: false,
+      babelOptions: {
+        configFile: require.resolve(
+          "@babel/eslint-shared-fixtures/config/babel.config.js",
+        ),
+      },
+      ...overrideConfig?.parserOptions,
+    },
+  });
+
+  if (messages.length !== expectedMessages.length) {
+    throw new Error(
+      `Expected ${expectedMessages.length} message(s), got ${
+        messages.length
+      }\n${JSON.stringify(messages, null, 2)}`,
+    );
+  }
+
+  messages.forEach((message, i) => {
+    const formattedMessage = `${message.line}:${message.column} ${
+      message.message
+    }${message.ruleId ? ` ${message.ruleId}` : ""}`;
+    const expectedMessage = expectedMessages[i];
+
+    if (expectedMessage instanceof RegExp) {
```

Comment:
Added the ability to pass this function a `RegExp` so we can enable the `xit`ed test below.

Reply:
Can we use `jest`'s methods instead of writing our own errors checking logic?

Reply:
I think that would be great. Mind if we do that in a follow up PR?

Reply:
It's ok!